app/src/main/java/com/example/erabook/fragments/login/LoginFragment.kt

```diff
@@ -63,46 +110,33 @@ class LoginFragment : Fragment() {
                 if (passwordLogIn.isEmpty()) {
                     requireContext().showToast(R.string.enter_password)
                 }
-                auth.signInWithEmailAndPassword(emailLogIn.trim(), passwordLogIn)
-                    .addOnCompleteListener { task ->
-                        if (task.isSuccessful) {
-                            startActivity(
-                                Intent(
-                                    requireContext(),
-                                    MainActivity::class.java
-                                ).apply {
-                                    putExtra(DESTINATION_PROFILE, R.id.userProfileFragment)
-                                })
-                            requireContext().showToast(R.string.login_successful)
-                        } else {
-                            val exception = task.exception as FirebaseException
-                            exception.let {
-                                when (it.message) {
-                                    getString(R.string.error_code_message_incorrect) -> requireContext().showToast(
-                                        R.string.wrong_password_email
-                                    )
+                authenticationViewModel.signInWithEmailAndPassword(emailLogIn.trim(), passwordLogIn)
+                authenticationViewModel.isTaskSuccessful.observe(viewLifecycleOwner) { task ->
+                    if (task == true) {
+                        findNavController().navigate(R.id.loginToProfile)
+                        requireContext().showToast(R.string.login_successful)
+                    } else {
+                        authenticationViewModel.exception.let { message ->
```

Comment:
Check let scope function: [https://kotlinlang.org/docs/scope-functions.html#let](https://kotlinlang.org/docs/scope-functions.html#let)

if you are using let here for nullability then you're not triggering it well. exception?.let is the right one.

Also if exception is null then the user will now see any message to what happened based on this implementation.

Suggestion: use elvis operator to handle the default message if exception is null.